examples/repl/repl.cpp

```diff
@@ -0,0 +1,81 @@
+#include <kalcy/kalcy.hpp>
+#include <cassert>
+#include <format>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+namespace {
+struct Repl {
+	bool verbose{}; // toggled on with '-v' and off with '-q'
+	bool is_running{true};
+	kalcy::Parser parser{};
+
+	auto  print_help() const -> void {
+		std::cout << "Usage: [OPTION] | [EXPRESSION]\n";
+		std::cout << "\nDescription:\n\n";
+		std::cout << std::format("  {:<15} {}\n", "-h, -help", "Display this help message, providing information about available options.");
+		std::cout << std::format("  {:<15} {}\n", "-v", "Toggle verbose mode to control the level of detail in the output.");
+		std::cout << std::format("  {:<15} {}\n\n", "exit", "Terminate the REPL input loop.");
+	}
+
+	auto start() -> void {
+		while (is_running) {
+			auto text = std::string{};
+			std::cout << std::format("{} > ", verbose ? "[verbose]" : "");
+			std::getline(std::cin, text);
+			// run kalcy on input expression
+			if (!text.empty()) {
+				run(text);
+			} else {
+				print_help();
+			}
+		}
+		// print epilogue
+		std::cout << std::format("\n^^ kalcy v{}\n", kalcy::version_v);
+	}
+
+	auto run(std::string_view const text) -> bool {
+		try {
+			if (text == "exit") {
+				is_running = false;
+				return true;
+			}
+
+			if (text == "-h" || text == "-help") {
```

Comment:
The long form of options is usually preceded by two dashes: `--help`, `--version`, etc.